lib/deno_ex.ex

```diff
@@ -90,11 +90,58 @@ defmodule DenoEx do
     end)
   end
 
+  @doc """
+  Vendors Deno dependencies into the given location
+  """
+  def vendor_dependencies(script_paths, vendor_location, lock_file_location, args) do
+    deno_path = Path.join(DenoEx.executable_path(), "deno")
+
+    System.cmd(
+      deno_path,
+      ~w[vendor #{Enum.join(script_paths, " ")} --output #{vendor_location} --lock=#{lock_file_location}] ++ args
+    )
+  end
+
+  @doc """
+  Locks the Deno dependencies
+  """
+  def lock_dependencies(script_paths, lock_file_location, _args) do
+    deno_path = Path.join(DenoEx.executable_path(), "deno")
+
+    Enum.each(script_paths, fn script_path ->
+      System.cmd(deno_path, ~w[cache --lock=#{lock_file_location} #{script_path}])
```

Comment:
Should we check to see that only files and not directories are processed here?

Reply:
I debated that, but should we also check for all the file types allowed to be run, like `ts`, `js`, `wasm`? I was torn on if it was worth it or if we should let the underlying deno error message come through. I could go either way. The deno error messages aren't great.